### PR TITLE
Jest components tests

### PR DESCRIPTION
**Pull request by ylavoie**
This PR adds the basis for Jest tests for our UI and allow testing it independently from the server.
 
The infrastructure comprises:
- JSDOM to allow testing in a browserlike environment.
- A Mock Service Worker (MSW) to mock at network in order to fully test our components and code completely and without any changes.
- Mocks of the WebElements (lsmb-etxt, lsmb-password, ...) in order to free ourselves from Dojo & Dijit
 
Store tests and LoginPage as set first tests.

### CODE REVIEW

**Review by ehuelsmann — CHANGES_REQUESTED** [review]

File: UI/src/views/LoginPage.vue [hunk 1]
```diff
@@ -70,16 +71,14 @@
...
 import { defineComponent, ref } from "vue";
 import { useI18n } from "vue-i18n";
 import { setI18nLanguage } from "@/i18n";
 import { createLoginMachine } from "./LoginPage.machines.js";
 
-
 export default defineComponent({
     name: "LoginPage",
-    setup(props, context) {
+    setup(props, context) { // eslint-disable-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply UI/src/views/LoginPage.vue:81]: Neither props not context

File: UI/src/views/LoginPage.vue [hunk 2]
```diff
@@ -43,8 +41,9 @@
                                      @keyup.enter="machine.send('submit')"
                                      @input="update"
                       />
-                      <lsmb-text name="company"
-                                 id="company" size="20"
+                      <lsmb-text
+id="company"
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply UI/src/views/LoginPage.vue]: Fixed

File: UI/src/views/LoginPage.vue [hunk 3]
```diff
@@ -53,11 +52,13 @@
                 </div>
                 <lsmb-button
                     v-if="state !== 'failed'"
-                    tabindex="4"
                     id="login"
-                    @click="machine.send('submit')"
-                    :disabled="state !== 'ready'">{{ $t('Login') }}</lsmb-button>
-                <div v-else style="background-color:red;border-radius:3px;padding:3px;font-weight:bold">{{ errorText }}</div>
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply UI/src/views/LoginPage.vue:60]: Fixed

File: UI/src/views/LoginPage.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (ylavoie) [reply UI/src/views/LoginPage.vue]: Fixed

**Review by ehuelsmann — APPROVED** [review]
Thanks for the efforts.

**Pull request merged**